src/main/java/org/jenkinsci/main/modules/sshd/SSHD.java

```diff
@@ -158,8 +159,14 @@ public synchronized void start() throws IOException, InterruptedException {
             try {
                 // In sshd-core 0.8.0 it must be an int
                 // https://github.com/apache/mina-sshd/blob/sshd-0.8.0/sshd-core/src/main/java/org/apache/sshd/server/session/ServerSession.java#L68
-                Integer.parseInt(idleTimeout);
-                sshd.getProperties().put(org.apache.sshd.server.ServerFactoryManager.IDLE_TIMEOUT, idleTimeout);
+                int idleTimeoutAsInt = Integer.parseInt(idleTimeout);
+                sshd.getProperties().put(ServerFactoryManager.IDLE_TIMEOUT, idleTimeoutAsInt);
+                // Read timeout must also be changed
+                long readTimeout = 0;
+                if (idleTimeoutAsInt != 0) {
+                    readTimeout = ServerFactoryManager.DEFAULT_NIO2_READ_TIMEOUT - ServerFactoryManager.DEFAULT_IDLE_TIMEOUT + idleTimeoutAsInt;
+                }
+                sshd.getProperties().put(ServerFactoryManager.NIO2_READ_TIMEOUT, readTimeout);
```

Comment:
Weird, I thought I had put a comment here. So I it seems like this `idleTimout` value computation could be easily extracted in a method, and this method be tested in various ways to make sure this is correctly working.

Reply:
I extracted an IdleTimeout class and added tests for it.